TimberAPIExample/Examples/EntityLinkerExample/UI/StartLinkingButton.cs

```diff
@@ -0,0 +1,114 @@
+        protected readonly ToolManager _toolManager;
+        protected Button _button;
+
+        public StartLinkingButton(
+            ILoc loc,
+            PickObjectTool pickObjectTool,
+            SelectionManager selectionManager,
+            ToolManager toolManager)
+        {
+            _loc = loc;
+            _pickObjectTool = pickObjectTool;
+            _selectionManager = selectionManager;
+            _toolManager = toolManager;
+        }
+
+        public virtual void Initialize<T>(VisualElement root,
+                                       Func<EntityLinker> linkerProvider,
+                                       Action createdLinkCallback)
+            where T : MonoBehaviour, IRegisteredComponent
+        {
+            _button = root.Q<Button>("NewLinkButton");
+            _button.clicked += delegate
+            {
+                StartLinkEntities<T>(linkerProvider(), createdLinkCallback);
+            };
+        }
+
+        /// <summary>
+        /// Fires up the object picker tool to select the linkee.
+        /// Called when the button is pressed
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="linker"></param>
+        /// <param name="createdLinkCallback"></param>
+        protected virtual void StartLinkEntities<T>(EntityLinker linker,
+                                                    Action createdLinkCallback)
+            where T : MonoBehaviour, IRegisteredComponent
+        {
+            _pickObjectTool.StartPicking<T>(
+                _loc.T(StartLinkingTitleLocKey),
+                _loc.T(StartLinkingTipLocKey),
+                (GameObject gameobject) => ValidateLinkee(linker, gameobject),
+                delegate (GameObject linkee)
+                {
+                    FinishLinkSelection(linker, linkee, createdLinkCallback);
+                });
+        }
+
+        /// <summary>
+        /// Validation logic for the linkee. Return empty string if valid.
+        /// Used for example if the entities need to be connected with a path. 
+        /// </summary>
+        /// <param name="linker"></param>
+        /// <param name="gameObject"></param>
+        /// <returns></returns>
+        protected virtual string ValidateLinkee(
+            EntityLinker linker,
+            GameObject gameObject)
+        {
+            return "";
+        }
+
+        /// <summary>
+        /// Creates a link between entities after the selection
+        /// </summary>
+        /// <param name="linker"></param>
+        /// <param name="linkee"></param>
+        /// <param name="createdLinkCallback"></param>
+        protected virtual void FinishLinkSelection(
+            EntityLinker linker,
+            GameObject linkee,
+            Action createdLinkCallback)
+        {
+            EntityLinker linkeeComponent = linkee.GetComponent<EntityLinker>();
+            linker.CreateLink(linkeeComponent);
```

Comment:
This is a fantastic example of how to establish a link, is there a way we can demonstrate how to use one? (that doesn't overlap your mod - something simple)

Reply:
@hawkfalcon That's a good idea. Will try to do a simple and smart enough example